test/integration/etcd/data.go

```diff
@@ -322,6 +333,16 @@ func GetEtcdStorageDataForNamespace(namespace string) map[schema.GroupVersionRes
 		},
 		// --
 
+		// k8s.io/kubernetes/pkg/apis/storage/v1beta1
+		gvr("storage.k8s.io", "v1beta1", "csistoragecapacities"): {
+			Stub:             `{"metadata": {"name": "csc-12345-2"}, "storageClassName": "sc1"}`,
+			ExpectedEtcdPath: "/registry/csistoragecapacities/" + namespace + "/csc-12345-2",
+			// TODO: should this be beta1?
+			// alpha is what makes `go test k8s.io/kubernetes/test/integration/etcd` pass.
+			ExpectedGVK: gvkP("storage.k8s.io", "v1alpha1", "CSIStorageCapacity"),
+		},
```

Comment:
I could use some guidance. What is the right value for `ExpectedGVK` here? It passes `go test k8s.io/kubernetes/test/integration/etcd` when using "v1alpha1", but that seems counter-intuitive. Do I perhaps have some missing code or a bug elsewhere such that "v1beta1" is not used?

When bringing up a cluster, the v1beta1 API works as expected. A writer which stores CSIStorageCapacity objects using v1alpha1 works and then `kubect get csistoragecapacities` returns the objects as v1beta1.

Reply:
Yes, this means the data is still being stored in etcd in v1alpha1.

`NewStorageFactoryConfig()` is explicitly overriding the preferred storage version for this type to be v1alpha1.

Reply:
Are these overrides in `NewStorageFactoryConfig()` needed? I probably added the one for `csistoragecapacities` only because there was one for `csidrivers`, something that got removed since then.

Reply:
yes. v1 is the preferred etcd version for storage.k8s.io, but this type does not exist in that version yet.

Reply:
Looks like these overrides are necessary for all (?) APis which are not yet in v1. I'll change the existing entry for "csistoragecapacities" to "v1beta1"